### PR TITLE
Move runtime libraries from iree/tools/ into runtime/.

### PR DESCRIPTION
**Pull request by ScottTodd**
Progress on https://github.com/google/iree/issues/8955, follow-up to https://github.com/google/iree/pull/9077.

This moves runtime library code from `iree/tools/utils` to `runtime/src/iree/tools/utils/`. The 'utils' name is still unfortunate (preferably each file would have a better place), but we can address that further down the line.

### CODE REVIEW

**Comment by benvanik** [comment]
let's keep this under a runtime/...iree/tools/utils/ dir instead of just iree/utils/ - we don't want anyone using these (they are utils just for tools, not for the runtime or for users of its APIs)

File: CMakeLists.txt [hunk 1]
```diff
@@ -690,14 +693,10 @@ set(IREE_PUBLIC_INCLUDE_DIRS "${IREE_COMMON_INCLUDE_DIRS}"
...
 #-------------------------------------------------------------------------------
-# Transitional directories.
+# Samples
 #-------------------------------------------------------------------------------
 
-# The runtime uses tools and targets from elsewhere in the tree in a way
-# that is not supported by CMake, so we include it last (it aggregates
-# many things into top-level APIs). This should resolve once we finish
-# relayering everything.
-add_subdirectory(runtime)
```
Comment (ScottTodd) [inline CMakeLists.txt:700]: Hm.. need to look a bit more closely at this for the `runtime/bindings/python/` package:

https://github.com/google/iree/blob/f56b906312b1b621ec86e3c96aabddd2325a0f40/runtime/bindings/python/CMakeLists.txt#L196-L204

```
CMake Error at runtime/bindings/python/CMakeLists.txt:196 (install):
  install TARGETS given target "iree_tools_iree-benchmark-module" which does
  not exist.
```

https://source.cloud.google.com/results/invocations/784aeae7-983e-4832-b10b-416aed15da9f/targets/iree%2Fgcp_ubuntu%2Fcmake%2Flinux%2Fx86-swiftshader%2Fpresubmit/log

File: runtime/src/iree/tools/utils/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ################################################################################
 # Autogenerated by build_tools/bazel_to_cmake/bazel_to_cmake.py from           #
-# iree/tools/utils/BUILD                                                       #
+# runtime/src/iree/tools/utils/BUILD                                           #
```
Comment (ScottTodd) [inline runtime/src/iree/tools/utils/CMakeLists.txt:3]: The `cmake_baremetal_riscv32` build is currently relying on this early exit turning off the `yaml` target (used via the `yaml_util` and `trace_replay` libraries):

https://github.com/google/iree/blob/500e41138cfa47b535de33a34c5d3de9eae25662/iree/tools/CMakeLists.txt#L9-L13

[error logs here](https://source.cloud.google.com/results/invocations/88920622-5240-4063-9360-e5e325a39090/targets/iree%2Fgcp_ubuntu%2Fcmake%2Fbaremetal%2Friscv32%2Fpresubmit/log):
```c
../third_party/libyaml/src/api.c:66:27: error: implicit declaration of function 'strdup' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    return (yaml_char_t *)strdup((char *)str);
                          ^
../third_party/libyaml/src/api.c:66:27: note: did you mean 'strcmp'?
/usr/src/toolchain_iree_rv32imf/bin/../riscv32-unknown-elf/include/string.h:36:6: note: 'strcmp' declared here
int      strcmp (const char *, const char *);
         ^
```

File: CMakeLists.txt (hunk 1 above)
Comment (ScottTodd) [reply CMakeLists.txt]: That will need a bit more work to move around. I just reverted my changes to subdirectory include order to fix it.

**Review by GMNGeoffrey — APPROVED** [review]

**Pull request merged**